### PR TITLE
[Ceph] Add Integration Package with OSD Performance data stream

### PR DESCRIPTION
**Pull request by harnish-elastic**
- Enhancement

## What does this PR do?

- Added 1 data stream (OSD Performance metrics).
- Added data collection logic for the data streams.
- Added the ingest pipeline for the data streams.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yaml files.
- Added system test cases for the data stream.

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's changelog.yml file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

- Clone integrations repo.
- Install elastic-package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/ceph directory.
- Run the following command to run tests.
elastic-package test

## Related issues

- Relates #4871
- Relates #4963
- Relates #5011 
- Relates #5070
- Relates #5128 
- Relates #5152
- Relates #5174

### CODE REVIEW

**Review by kush-elastic — APPROVED** [review]
LGTM!

**Comment by SubhrataK** [comment]
Please share the readme in a doc file with me separately. Otherwise this looks good to me.

**Review by SubhrataK — APPROVED** [review]

File: packages/ceph/data_stream/osd_performance/_dev/test/pipeline/test-osd-performance-metrics.log-expected.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
+    "expected": [
+        {
+            "ceph": {
+                "osd_performance": {
+                    "id": 2,
```
Comment (agithomas) [inline packages/ceph/data_stream/osd_performance/_dev/test/pipeline/test-osd-performance-metrics.log-expected.json:6]: Please change to osd_id

Comment (harnish-elastic) [reply packages/ceph/data_stream/osd_performance/_dev/test/pipeline/test-osd-performance-metrics.log-expected.json:6]: In this data stream, we have `osd_performance` as parent field. So the current output will be like `ceph.osd_performance.id`. Based on the suggested scenario, we will have `ceph.osd_performance.id`. So I think this change would make osd redundant. WDYT?

Comment (agithomas) [reply packages/ceph/data_stream/osd_performance/_dev/test/pipeline/test-osd-performance-metrics.log-expected.json:6]: Yes please

**Review by agithomas — APPROVED** [review]
LGTM

File: packages/ceph/data_stream/osd_performance/fields/fields.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                  type: float
+                  description: Time taken to flush an update to disks. Collects in milliseconds.
+                  metric_type: gauge
+                  unit: ms
+            - name: commit
+              type: group
+              fields:
+                - name: ms
+                  type: float
+                  description: Time taken to commit an operation to the journal. Collects in milliseconds.
```
Comment (agithomas) [inline packages/ceph/data_stream/osd_performance/fields/fields.yml:23]: Please choose the right metric_type . Ref : https://www.elastic.co/guide/en/elasticsearch/reference/current/api-conventions.html#time-units

File: packages/ceph/data_stream/osd_performance/fields/fields.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        - name: latency
+          type: group
+          fields:
+            - name: apply
+              type: group
+              fields:
+                - name: ms
+                  type: float
+                  description: Time taken to flush an update to disks. Collects in milliseconds.
+                  metric_type: gauge
```
Comment (agithomas) [inline packages/ceph/data_stream/osd_performance/fields/fields.yml:16]: Please choose the right metric_type . Ref : https://www.elastic.co/guide/en/elasticsearch/reference/current/api-conventions.html#time-units

**Review by agithomas — APPROVED** [review]
LGTM!

**Pull request merged**